### PR TITLE
Add bgen

### PR DESCRIPTION
**Pull request by Horta**

### CODE REVIEW

**Comment by Horta** [comment]
Done =) Ready to add!

**Comment by Horta** [comment]
Bump!

File: recipes/bgen/bld.bat [hunk 1]
```diff
@@ -0,0 +1,15 @@
+mkdir build
+if errorlevel 1 exit 1
+cd build
+if errorlevel 1 exit 1
```
Comment (ocefpaf) [inline recipes/bgen/bld.bat]: You don't really need the `if errorlevel 1 exit 1` for creating and entering the directory.
If this fails, which is unlikely, the cake below will fail. (It does not hurt to leave it there though.)

File: recipes/bgen/bld.bat [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+cmake -G "NMake Makefiles" ^
+         -DCMAKE_WINDOWS_EXPORT_ALL_SYMBOLS=TRUE ^
+         -DCMAKE_INSTALL_PREFIX:PATH=%PREFIX% ^
+         ..
```
Comment (ocefpaf) [inline recipes/bgen/bld.bat]: Use the more robust `%SRC_DIR%` variable.

File: recipes/bgen/build.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+set -ex
+
+export CFLAGS="${CFLAGS} -O3 -fPIC"
+export LDFLAGS="${LDFLAGS} -Wl,-rpath,${PREFIX}/lib"
+
+mkdir build
```
Comment (ocefpaf) [inline recipes/bgen/build.sh]: You can save a few lines by doing `mkdir build && cd build`

File: recipes/bgen/build.sh [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+cd build
+
+cmake -DCMAKE_INSTALL_PREFIX:PATH=$PREFIX ..
```
Comment (ocefpaf) [inline recipes/bgen/build.sh]: You can also use `$SRC_DIR` here instead of `..`

File: recipes/bgen/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True  # [win32 or linux32 or (win64 and py27)]
+
+requirements:
+  build:
+    - python  # [win]
+    - conda-build
```
Comment (ocefpaf) [inline recipes/bgen/meta.yaml]: This is odd. Do you need to specify `conda-build` here? It is granted that `conda-build` will be available anyways.

File: recipes/bgen/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    - toolchain
+    - cmake
+    - zstd
+    - zlib  # [win]
```
Comment (ocefpaf) [inline recipes/bgen/meta.yaml]: Please add the pinning.

File: recipes/bgen/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  run:
+    - zstd
+    - zlib  # [win]
+
+test:
+  requires:
+    - python  # [win]
```
Comment (ocefpaf) [inline recipes/bgen/meta.yaml]: You actually need this in the `run` section. Also you need to add the `vc` features.
See http://conda-forge.github.io/docs/buildwin.html#vc-features

File: recipes/bgen/meta.yaml (hunk 6 above)
Comment (Horta) [inline recipes/bgen/meta.yaml]: This one I didn't get... what do you mean?

File: recipes/bgen/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    - cmake
+    - zstd
+    - zlib  # [win]
+  run:
+    - zstd
+    - zlib  # [win]
+
+test:
+  requires:
+    - python {{ environ['PY_VER'] + '*' }}  # [win]
```
Comment (ocefpaf) [inline recipes/bgen/meta.yaml]: If you add the `vc`s package on the `run` and `build` you don't need this bad hack.

Comment (Horta) [inline recipes/bgen/meta.yaml]: I'm very confused... shall I just remove those lines?

Comment (ocefpaf) [inline recipes/bgen/meta.yaml]: You need to add

```yaml
    - vc 9  # [win and py27]
    - vc 10  # [win and py34]
    - vc 14  # [win and (py35 or py36)]
```

to both build and run time dependencies so the right `vc` is installed when you are using the package.
With that you can, and should, remove the hack:

```yaml
 requires:
    - python {{ environ['PY_VER'] + '*' }}  # [win]
```

Comment (ocefpaf) [inline recipes/bgen/meta.yaml]: You missed the space. The `vc` package is not the same as the vc feature. Confusing, I know...

`vc 14` installs the feature `vc14`.

**Comment by Horta** [comment]
Good to go?

**Pull request merged**